Applications/MNIST/README.md

```diff
@@ -0,0 +1,141 @@
+# MNIST
+
+In this example, we demonstrate training full neural network model with mnist dataset.
+
+### Neural Network Model Configuration
+Network model consists of two convolution layer, two pooling layer, one flatten layer and one fully connected layer.
+
+<p align = "center">
+<img src="https://github.com/nnstreamer/nntrainer/blob/master/docs/images/mnist_model.png">
+</p>
+
+
+### MNIST Data Set and DataBuffer
+In order to make training short, we use subset of full mnist dataset, 100 images per class and save it into file named ```mnist_trainingSet.dat```. The input image size is 784 (28x28) and the label is an one-hot vector for classifying 10 digits. The ith image feature and label data must be at ```i x ((784 x sizeof(float) + label(10 x sizeof(float))``` byte position. 
+
+``` bash
+...
+0 0 0 223 225 0 0 0 0 0 0... 0  --> ith feature data (784xsizeof(float))
+1 0 0 0 0 0 0 0 0 0         --> ith label data as an one-hot vector
+0 0 0 0 0 0 0 0 224 123 0... 0  --> i+1th feature data (784xsizeof(float))
+0 1 0 0 0 0 0 0 0 0         --> i+1th label data as an one-hot vector
+...
+
+```
+
+In order to get the training data, ```DataBufferFromCallback``` option is used and ```getData_train``` and ```getData_val``` functions are defined which reads the feature and label with batch_size and fill the databuffer. If there is no data for an epoch, then ```last``` will be true.
```

Comment:
If there is no data for an epoch -> If there is no `remaining` data for `the current` epoch